server/src/main/java/com/adobe/testing/s3mock/FileStoreController.java

```diff
@@ -705,11 +717,59 @@ public void getObject(@PathVariable final String bucketName,
       response.setHeader(HttpHeaders.ACCESS_CONTROL_ALLOW_ORIGIN, ANY);
       response.setDateHeader(HttpHeaders.LAST_MODIFIED, s3Object.getLastModified());
       addUserMetadata(response::addHeader, s3Object);
+      addOverrideHeaders(response, request.getQueryString());
 
       try (final OutputStream outputStream = response.getOutputStream()) {
         Files.copy(s3Object.getDataFile().toPath(), outputStream);
       }
     }
+
+  }
+
+  private void addOverrideHeaders(final HttpServletResponse response, final String query) {
+    if (query != null && !query.isEmpty()) {
+      Arrays.stream(query.split("&"))
+          .map(this::splitQueryParameter)
+          .forEach((h) -> addOverrideHeader(response, h.getKey(), h.getValue()));
+    }
+  }
+
+  private SimpleImmutableEntry<String, String> splitQueryParameter(final String it) {
+    final int idx = it.indexOf("=");
+    try {
+      final String key = URLDecoder.decode(idx > 0 ? it.substring(0, idx) : it, UTF_8.name());
+      final String value = idx > 0 && it.length() > idx + 1
+          ? URLDecoder.decode(it.substring(idx + 1), UTF_8.name()) : null;
+      return new SimpleImmutableEntry<>(key, value);
+    } catch (UnsupportedEncodingException e) {
+      throw new AssertionError(UTF_8.name() + " is unknown");
+    }
+  }
+
+  private void addOverrideHeader(final HttpServletResponse response, final String name,
+      final String value) {
+    switch (name) {
+      case RESPONSE_HEADER_CACHE_CONTROL:
+        response.setHeader(HttpHeaders.CACHE_CONTROL, value);
+        break;
+      case RESPONSE_HEADER_CONTENT_DISPOSITION:
+        response.setHeader(HttpHeaders.CONTENT_DISPOSITION, value);
+        break;
+      case RESPONSE_HEADER_CONTENT_ENCODING:
+        response.setHeader(HttpHeaders.CONTENT_ENCODING, value);
+        break;
+      case RESPONSE_HEADER_CONTENT_LANGUAGE:
+        response.setHeader(HttpHeaders.CONTENT_LANGUAGE, value);
+        break;
+      case RESPONSE_HEADER_CONTENT_TYPE:
+        response.setContentType(value);
+        break;
+      case RESPONSE_HEADER_EXPIRES:
+        response.setHeader(HttpHeaders.EXPIRES, value);
+        break;
+      default:
+        // Only the above header overrides are supported by S3
```

Comment:
Add some logging here pls

Reply:
As it iterates over all query params and just filters out the known ones, should the unrelated query params really be logged? I'd say that's expected behavior and shouldn't be logged.

Reply:
mh, of course it comes with the potential to pollute logs ;) Alright, let's leave it as it is